financial-connections/src/main/java/com/stripe/android/financialconnections/ui/FinancialConnectionsSheetNativeActivity.kt

```diff
@@ -325,3 +359,37 @@ internal val LocalReducedBranding = staticCompositionLocalOf<Boolean> {
 internal val LocalImageLoader = staticCompositionLocalOf<StripeImageLoader> {
     error("No ImageLoader provided")
 }
+
+/**
+ * Observer that will notify the view model when the activity is moved to the background or
+ * brought back to the foreground.
+ */
+private class ActivityVisibilityObserver(
+    val onBackgrounded: () -> Unit,
+    val onForegrounded: () -> Unit
+) : DefaultLifecycleObserver {
+
+    private var isFirstStart = true
+    private var isInBackground = false
```

Comment:
Any concern with these being reset on configuration changes? My understanding is that the containing `DisposableEffect` will be disposed on config changes, which would mean that this object is being destroyed and re-created.

Reply:
yes, that's the case - the observer gets re-registered on config changes, and still behaves as intended:

- onStop -> it specifically checks if config changes are happening to prevent logging the background event
- onStart -> `isFirstStart = true` (since activity recreated) -> we won't log the foreground event

we could potentially save the `isFirstStart` field outside of the lifecycle as a `rememberSaveable` and pass it as a parameter to the observer to make this more robust, wdyt?

Reply:
Oh right, just tested it and it behaves correctly. I’m fine keeping it as-is then.